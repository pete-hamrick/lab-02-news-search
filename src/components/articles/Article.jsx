import React from 'react';

const Article = ({ author, description, title, url, urlToImage }) => (
  <article>
    <h2>{title}</h2>
    <img src={urlToImage} alt={title} style={{ height: 200 }}></img>
    <h4>{author}</h4>
    <p>{description}</p>
    <a href={url}>Read The Article</a>
  </article>
);

export default Article;
