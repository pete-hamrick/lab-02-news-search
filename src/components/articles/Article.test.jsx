// snapshot test
import React from 'react';
import Article from './Article';
import { render } from '@testing-library/react';

describe('Article Component', () => {
  it('Displays an article', () => {
    const { asFragment } = render(
      <Article
        author="Reporter Jones"
        description="The article summary"
        title="The Article Title"
        url="https://article.com/article"
        urlToImage="https://article.com/article.jpeg"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
