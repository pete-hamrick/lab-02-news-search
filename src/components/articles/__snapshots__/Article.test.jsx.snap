// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Component Displays an article 1`] = `
<DocumentFragment>
  <article>
    <h2>
      The Article Title
    </h2>
    <img
      alt="The Article Title"
      src="https://article.com/article.jpeg"
      style="height: 200px;"
    />
    <h4>
      Reporter Jones
    </h4>
    <p>
      Description
    </p>
    <p>
      The article summary
    </p>
    <a
      href="https://article.com/article"
    >
      Read The Article
    </a>
  </article>
</DocumentFragment>
`;
